### ISSUE TITLE
COM error [E_ABORT & CO_E_SERVER_EXEC_FAILURE]

### ISSUE THREAD

**Issue by BMBoyd** [phw198/OutlookGoogleCalendarSync#268]
Installing on Win 10 with OL13. Tried the CodePlex oneclick and portable versions here.
[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1135977/OGcalsync.txt)

**Comment by dutch55** [comment]
UPDATE: I BELIEVE THIS IS SAME AS ISSUE #263 
COM Exception on Windows 10 Preview 16237, Outlook 2016 -64bit  all updates.  

OGSC Build 2.5.1.0 Alpha

Error occured after reboot before Outlook or any other programs were launched from the keyboard.

[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1140097/OGcalsync.txt)

**Comment by DGreenebaum** [comment]
COM exception on Windows 7 Professional, Service Pack 1
Outlook 2013 (15.0.4937.1000) 32-bit, part of MS Office Professional Plus 2013

[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1142920/OGcalsync.txt)

**Comment by pitaki** [comment]
My issue #270 seems to be on this very subject too.

**Comment by phw198** [comment]
@lsu2010 
The error you're getting for the installed version is `Operation aborted (Exception from HRESULT: 0x80004004 (E_ABORT))`
..and for the portable version `Server execution failed (Exception from HRESULT: 0x80080005 (CO_E_SERVER_EXEC_FAILURE)`

Do you have Outlook running with elevated permissions (ie Run as Administrator), or are you running OGCS on a server / remote VM?

@dutch55 @DGreenebaum 
I can't see any error in your log file. Does it only happen on login to Windows as I see you have OGCS set to start on startup? If so, this matches #263

@pitaki 
Wouldn't be able to tell - you didn't provide a log file on #270. Also, if you think it's the same issue, don't raise another ticket.

**Comment by phw198** [comment]
I can replicate `CO_E_SERVER_EXEC_FAILURE` by running OGCS with Administrator privileges and Outlook as normal. 

@lsu2010 Can you confirm if you are running OGCS at a different security elevation to Outlook?

**Comment by pitaki** [comment]
Not to my knowledge. Both OGCS and Outlook are
triggered to go during boot and OCGS seem to
start first.

Aki

**Comment by phw198** [comment]
@pitaki start-up issues are on #263 with a delay option added in v2.5.2

**Comment by BMBoyd** [comment]
logs with and without admin privileges for just OCGS and both OCGS and Outlook. Looks like separate errors for the different scenarios. Running OGCS on local machine.
[OGcalsync_reg privileges.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1152961/OGcalsync_reg.privileges.txt)
[OGcalsync_admin privileges.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1152962/OGcalsync_admin.privileges.txt)
[OGcalsync_admin for OGCS and OL.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1153013/OGcalsync_admin.for.OGCS.and.OL.txt)


.

**Comment by phw198** [comment]
:package: So [v2.5.2.8.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1168450/v2.5.2.8.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)) should handle the different security elevations better. Could you let me know how it goes? 

Also, on a hunch, I've also added in a delay during the connection to Outlook process....you never know!

- Do you have any anti-virus running? Could you try disabling it and see if it makes any difference?
- What are your Trust Center Settings in Outlook under "Programmatic Access" and "Macro Settings"?

Test script & results thus far:
#### OGCS Install  

|Tested|OGCS Privs|Outlook Privs|Error|
|---|----|-----|----|
|Y|Regular|Regular|`E_ABORT`|
|Y|Admin|Regular|`CO_E_SERVER_EXEC_FAILURE`|
|Y|Admin|Admin|`E_ABORT`|

#### OGCS Portable  
Also, when you ran the portable version before and got `CO_E_SERVER_EXEC_FAILURE`, was that running both without admin privileges? Could you try the attached hotfix over the top of a portable deployment too, and provide the resulting logs?

|Tested|OGCS Privs|Outlook Privs|Error|
|---|----|-----|----|
|?|Regular|Regular|`CO_E_SERVER_EXEC_FAILURE`|
|N|Admin|Regular|?|
|N|Admin|Admin|?|

Appreciate your help investigating this :+1:

**Comment by pitaki** [comment]
Cheers.

Not much luck with these patches. Here’s the log-file after running that last one.

Somehow my credentials from Google seem to have been misplaced or something.

I do have a F-secure Safe -virusscanning going on and taking it off didn’t change anything. Not now, not before.

About those settings…
Progr..Access

All entries are grayed, though “Warn about suspicious activities when antivir is not in use or un-updated”
Antivirus program status = Accepted

Macros

Announce all digitally signed macros. All other off

Use macro protect settings also in installed auxiliary programs (not ticked)

Yours,
Aki

**Comment by phw198** [comment]
@pitaki Thanks, but
- no log file attached. Again, GitHub doesn't support updating tickets with attachments through email
- I said your issue is #263, which is fixed, so update that ticket if you don't think it is fixed.
- not sure why you're still following this issue (#268); it's rather confusing things

**Comment by pitaki** [comment]
[OGcalsync.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1172460/OGcalsync.zip)

**Comment by pitaki** [comment]
Sorry, Paul.

I’m not accustomed to this issue -system you use.
I didn’t find any “Discard” -button or anything from
these  issue-numbers, so can you point out the proper
procedure for me.

Another thing is that I tend to answer emails as they are
and these restrictions in recipients end are not always
considered due to that fact.

**Comment by BMBoyd** [comment]
@phw198 

All OL trust center settings grayed out. McAfee suite is running. Cannot disable. Was running when previous OGCS versions were running successfully (Windows 7 Office 2010). 

My previous work computer was Windows 7 Office 2010 - OGCS functioned fine. I think my issues could be due to different permissions and security settings by network administrators in W10 Office 2013. Thoughts?

#### OGCS Install Hotfix

| Tested | OGCS Privs | Outlook Privs | Error |
|---|---|---|---|
| Y | Regular | Regular | E_ABORT & popup - "Object reference not set to an instance of an object." 
Y | Admin | Regular | E_ABORT & popup - "Object reference not set to an instance of an object."
Y | Admin | Admin | E_ABORT & popup - "Object reference not set to an instance of an object."

#### OGCS Port Hotfix
Tested | OGCS Privs | Outlook Privs | Error
|---|---|---|---|
Y | Regular | Regular | E_ABORT & popup - "Object reference not set to an instance of an object."
Y | Admin | Regular | E_ABORT & popup - "Object reference not set to an instance of an object."
Y | Admin | Admin | E_ABORT & popup - "Object reference not set to an instance of an object."

Logs in order as above.
[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1173827/OGcalsync.txt)

**Comment by phw198** [comment]
@pitaki Your latest logfile is showing an Aggregate exception which you already raised under issue #270 and to which I posted a hotfix, but you have not replied.

@lsu2010 Thanks for the work on this. The exception handling is a little screwy, but it definitely looks like something is stopping OGCS from talking to Outlook. The closest you're able to get is when running Outlook and OGCS as administrator, but it's still blocked. Does feel like something has changed at your work.

You say that the options are greyed out in the Security Center, but which option is selected? Are you getting Outlook security pop-ups? If so are you clicking Allow or Deny? I've just done a bit of testing and can produce `E_ABORT` if I choose Deny...

**Comment by phw198** [comment]
So it seems that `CO_E_SERVER_EXEC_FAILURE` isn't only produced by differences in elevated permissions, but can also occur when Outlook is busy starting :-1: 

:package: Please patch up to the latest nightly build of [v2.5.2.12.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1195569/v2.5.2.12.zip) [[see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)].

**Comment by phw198** [comment]
@lsu2010 How is the latest beta v2.5.3 going for you? Has it fixed this issue?

**Comment by BMBoyd** [comment]
Issue remains. My guess is the connection is blocked by network
administrator or antivirus. Worked fine on previous OS and MS office
version. I have asked my IT contact and will let you know what I find.

**Comment by phw198** [comment]
Does v2.5.3 take you to issue #287? If not, could you post your OGcalsync.log file.

**Comment by BMBoyd** [comment]
@phw198 #287 hotfix is working for me! This makes my day so much easier. Thank you, thank you, thank you. I'll watch #287 for updates. Will post any issues there.
